File: content/04/code/js/11-replace-hardcoded-lucy-instance/test/network.test.js
const sinon = require('sinon')
const {Network} = require('../src/shouty')

describe('Network', () => {
  it('broadcasts a message to all listeners', () => {
    const network = new Network()
    const message = "Free bagels!"
    const lucy = { hear(message) {} }
    const lucyMock = sinon.mock(lucy)
    lucyMock.expects('hear').once().withArgs(message)
    network.subscribe(lucy)
    network.broadcast(message)
    lucyMock.verify()
  })
})

File: content/bdd-with-cucumber/04/code/java/01--step01-description-added/shouty/src/test/java/shouty/NetworkTest.java
package shouty;

import org.junit.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class NetworkTest {
    @Test
    public void broadcasts_a_message_to_all_listeners() {
        Network network = new Network();
        String message = "Free bagels!";
        Person lucy = mock(Person.class);
        network.subscribe(lucy);
        network.broadcast(message);
        verify(lucy).hear(message);
    }
}